presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/Decoders.java

```diff
@@ -178,6 +179,9 @@ private static ValuesDecoder createValuesDecoder(ColumnDescriptor columnDescript
                     if (isTimeStampMicrosType(columnDescriptor)) {
                         return new Int64TimestampMicrosRLEDictionaryValuesDecoder(bitWidth, inputStream, (LongDictionary) dictionary);
                     }
+                    if (isDecimalType(columnDescriptor) && isShortDecimalType(columnDescriptor)) {
```

Comment:
Can we omit the first condition? Is there a scenario where the `columnDescriptor` is a `ShortDecimalType` but not a `DecimalType`?

Reply:
Confusing, but these technically, these check for two separate things

- `isDecimalType` checks if the primitive type is `DECIMAL`
- `isShortDecimal` actually checks the logical type annotation is decimal and then gets the precision parameter from the logical type to check if it is a short decimal

From my understanding `isShortDecimal` should always be true if `isDecimalType` returns true. However, this is the same check as in the previous block for `FLOAT`. I would prefer to stay consistent. Also, it is my understanding that Presto may not properly write logical type annotations yet according to #23388 -- so maybe let's just keep this now for consistency's sake?

Reply:
If we look deeper at `isDecimalType`, we will find that it ultimately also verifies that the logical type annotation is decimal. So as I understand, if a `columnDescriptor` is of type `ShortDecimalType`, it must first be of type `DecimalType`. Furthermore, if the primitive type's logical type annotation is null, it can neither be a `DecimalType` nor a `ShortDecimalType`, so it won't break the conclusion above (The current experiments in my local show the same behavior). And we also can see that in `encoding == PLAIN` clause, the checks in `INT32` and `INT64` are all `isShortDecimalType` only.

I'm OK for now to keep it as is, since I am not very sure if there are still some special scenarios present. But once it is completely checked and confirmed, I think it would be better to delete the first condition (so as to the previous block you mentioned) because it would cause a lot of confusion for future readers.